### PR TITLE
[Merged by Bors] - feat: add some lemmas about Monoid.exponent and Int-powers

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/Exponent.lean
Comment (MichaelStollBayreuth) [inline Mathlib/GroupTheory/Exponent.lean:497]: Maybe instead add `variable {g : G} {n m : ℤ}` just before the new stuff?

File: Mathlib/GroupTheory/Exponent.lean
Comment (MichaelStollBayreuth) [inline Mathlib/GroupTheory/Exponent.lean:531]:
```suggestion
  Group.exponent_dvd_iff_forall_zpow_eq_one.mp h g
```
Why not use the new lemma?

File: Mathlib/GroupTheory/Exponent.lean
Comment (MichaelStollBayreuth) [inline Mathlib/GroupTheory/Exponent.lean:537]:
```suggestion
    (Int.natCast_dvd_natCast.mpr <| Monoid.order_dvd_exponent g).trans h
```

**Review by MichaelStollBayreuth — COMMENTED** [review]
LGTM, modulo some minor potential golfing.

**Comment by MichaelStollBayreuth** [comment]
maintainer delegate

File: Mathlib/GroupTheory/Exponent.lean
Comment (Ruben-VandeVelde) [reply Mathlib/GroupTheory/Exponent.lean]: Because apparently I got distracted and didn't finish what I started before pushing - I just meant to have the iffs.

**Review by MichaelStollBayreuth — APPROVED** [review]
OK!
maintainer merge

**Review by Vierkantor — APPROVED** [review]
Looks good to me, thanks!

bors merge

**Pull request closed without merging**
